### ISSUE TITLE
[New rule] prefer_list_first

### ISSUE THREAD

**Issue by roman-petrov** [dart-code-checker/dart-code-metrics#481]
**Please describe what the rule should do:**

During code review, I usually suggest using `List.first` instead of `List[0]` for better readability. Just noticed that I am [not alone here](https://github.com/dart-lang/linter/pull/2967/files). :)

**If your rule is inspired by other please provide link to it:**

Nope.

**What category of rule is this? (place an "X" next to just one item)**

[ ] Warns about a potential error (problem)
[x] Suggests an alternate way of doing something (suggestion)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule will warn about (it will be better if you can provide both good and bad examples):**

**BAD:**

```dart
final integers = [1, 2, 3];
final firstElement = integers[0];
```

**GOOD:**

```dart
final integers = [1, 2, 3];
final firstElement = integers.first;
```

**Are you willing to submit a pull request to implement this rule?**

Maybe.
